Rocky.class.nut

```diff
@@ -199,52 +200,71 @@ class Rocky {
         if (contentType.find("multipart/form-data;") == 0) {
             local parts = [];
 
-            // _parse_body is wrapped in a try/catch.. so we just let this fail
-            // when the content-type isn't long enough (and on other issues).
-            local boundary = contentType.slice(30);
-
-            local bindex = -1;
-            do {
-                bindex = req.body.find("--" + boundary + "\r\n", bindex+1);
-
-                if (bindex != null) {
-                    // Locate all the parts
-                    local hstart = bindex + boundary.len() + 4;
-                    local nstart = req.body.find("name=\"", hstart) + 6;
-                    local nfinish = req.body.find("\"", nstart);
-                    local fnstart = req.body.find("filename=\"", hstart) + 10;
-                    local fnfinish = req.body.find("\"", fnstart);
-                    local bstart = req.body.find("\r\n\r\n", hstart) + 4;
-                    local fstart = req.body.find("\r\n--" + boundary, bstart);
-
-                    // Pull out the parts as strings
-                    local headers = req.body.slice(hstart, bstart);
-                    local name = null;
-                    local filename = null;
-                    local type = null;
-                    foreach (header in split(headers, ";\n")) {
-                        local kv = split(header, ":=");
-                        if (kv.len() == 2) {
-                            switch (strip(kv[0]).tolower()) {
-                                case "name":
-                                    name = strip(kv[1]).slice(1, -1);
-                                    break;
-                                case "filename":
-                                    filename = strip(kv[1]).slice(1, -1);
-                                    break;
-                                case "content-type":
-                                    type = strip(kv[1]);
-                                    break;
-                            }
-                        }
+            // Find the boundary in the contentType
+            local boundary;
+            local findString = regexp(@"boundary([ ]*)=");
+            local match = findString.search(contentType);
+
+            if (match) {
+                boundary = contentType.slice(match.end);
+                boundary = strip(boundary);
+            } else {
+                throw "No boundary found in content-type";
+            }
+
+            // Remove all carriage returns from string (to support either \r\n or \n for linebreaks)
+            local body = "";
+            local bodyLines = split(req.body, "\r");
+            foreach (i, line in bodyLines) {
+                body += line;
+            }
+
+            // Find all boundaries in the body
+            local boundaries = [];
+            local bregex = regexp2(@"--" + boundary);
```

Comment:
Just curious, is there any specific reason to use regexp2? Doesn't regular regexp work here?

Reply:
I tested by cahnging all instances to `regexp()` but it didn't work. I had a look at the docs for the two functions but I'm not sure why it doesn't work. I'd rather keep them consistent and use only `regexp()` or `regexp2()`.

Reply:
makes sense, agree